From 95e42c51203e87c247d6d601c63b4dab0f52ba5d Mon Sep 17 00:00:00 2001
From: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Date: Fri, 3 Dec 2010 15:30:27 -0500
Subject: [PATCH] radeon/PCIe: Use the correct index field.

We were using the wrong index field causing us to ignore
when DMA addresses were passed in.

Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Signed-off-by: Ian Campbell <ian.campbell@citrix.com>
(cherry-picked from 8db353a504ed6bc4473e559cd3249830fae988aa in
git://git.kernel.org/pub/scm/linux/kernel/git/konrad/xen.git#devel/ttm.pci-api-v2)
---
 drivers/gpu/drm/radeon/radeon_gart.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/radeon/radeon_gart.c b/drivers/gpu/drm/radeon/radeon_gart.c
index f5d513d..2f01168 100644
--- a/drivers/gpu/drm/radeon/radeon_gart.c
+++ b/drivers/gpu/drm/radeon/radeon_gart.c
@@ -180,9 +180,9 @@ int radeon_gart_bind(struct radeon_device *rdev, unsigned offset,
 	for (i = 0; i < pages; i++, p++) {
 		/* On TTM path, we only use the DMA API if TTM_PAGE_FLAG_DMA32
 		 * is requested. */
-		if (dma_addr[p] != DMA_ERROR_CODE) {
+		if (dma_addr[i] != DMA_ERROR_CODE) {
 			rdev->gart.ttm_alloced[p] = true;
-			rdev->gart.pages_addr[p] = dma_addr[p];
+			rdev->gart.pages_addr[p] = dma_addr[i];
 		} else {
 			/* we need to support large memory configurations */
 			/* assume that unbind have already been call on the range */
-- 
1.5.6.5

